assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -437,8 +437,39 @@ describe("DiversionPage", () => {
       />
     )
 
-    expect(container.querySelectorAll(".c-stop-icon-container")).toHaveLength(
-      11
+    expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(11)
+  })
+
+  test("missed stop markers are drawn on the map", async () => {
+    const stop1 = stopFactory.build()
+    const stop2 = stopFactory.build()
+    const stop3 = stopFactory.build()
+    const stop4 = stopFactory.build()
+
+    jest
+      .mocked(fetchFinishedDetour)
+      .mockResolvedValue(finishedDetourFactory.build({ missedStops: [stop2] }))
+
+    const { container } = render(
+      <DiversionPage
+        originalRoute={{
+          shape: shapeFactory.build({ stops: [stop1, stop2, stop3, stop4] }),
+        }}
+      />
     )
+
+    await act(async () => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+    await act(async () => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    waitFor(() => {
+      expect(screen.getByText(stop2.name)).toBeInTheDocument()
+    })
+
+    expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(3)
+    expect(container.querySelectorAll(".c-missed-stop-icon")).toHaveLength(1)
   })
```

Comment:
question: Similar to https://github.com/mbta/skate/pull/2484#discussion_r1520288912 could this be simplified?
```suggestion
    waitFor(() => {
      expect(screen.getByText(stop2.name)).toBeInTheDocument()
      expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(3)
      expect(container.querySelectorAll(".c-missed-stop-icon")).toHaveLength(1)
    })
  })
```

I'm not sure if the first assertion is needed?
```suggestion
    waitFor(() => {
      expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(3)
      expect(container.querySelectorAll(".c-missed-stop-icon")).toHaveLength(1)
    })
  })
```

Reply:
[Yeah the first assertion isn't really what this test is about](https://github.com/mbta/skate/pull/2484/commits/c2163793b71d78153585c5fca0155197e64c28b3)